openupgradelib/openupgrade_merge_records.py

```diff
@@ -202,21 +202,38 @@ def _change_translations_orm(env, model_name, record_ids, target_record_id,
                              exclude_columns):
     if ('ir_translation', 'res_id') in exclude_columns:
         return
-    records = env['ir.translation'].search([
+    translation_obj = env['ir.translation']
+    groups = translation_obj.read_group([
         ('type', '=', 'model'),
         ('res_id', 'in', record_ids),
-        ('name', 'like', '%s,%%' % model_name)])
-    if records:
-        records.write({'res_id': target_record_id})
-        logger.debug(
-            "Changed %s translations of model 'ir.translation'",
-            len(records))
+        ('name', 'like', '%s,%%' % model_name),
+    ], ['name', 'lang'], ['name', 'lang'], lazy=False)
+    for group in groups:
+        target_translation = translation_obj.search([
+            ('type', '=', 'model'),
+            ('res_id', '=', target_record_id),
+            ('name', '=', group['name']),
+            ('lang', '=', group['lang']),
+        ])
+        records = translation_obj.search(group['__domain'])
+        if not records:
+            continue
+        if target_translation:
+            # A translation already exists - We delete the rest of translations
+            logger.debug("Deleted %s extra translations.'", len(records))
+            records.unlink()
+        else:
+            # There is no target translation, we pick one for being the new one
```

Comment:
Shouldn't you check if len(records) > 1? Because if len(records) = 1 then you shouldn't delete it.

Reply:
It doesn't happen anything. `unlink` over an empty recordset does nothing, but no error.

Reply:
Well, its a bit silly "Deleted 0 extra translations." Maybe this generates a lot of spam

Reply:
OK, changing directly in GitHub...